src/plugins/data/public/ui/dataset_selector/advanced_selector.tsx

```diff
@@ -13,19 +13,27 @@ import {
 } from '../../../common';
 import { DatasetExplorer } from './dataset_explorer';
 import { Configurator } from './configurator';
-import { getQueryService } from '../../services';
 import { IDataPluginServices } from '../../types';
 
 export const AdvancedSelector = ({
   services,
   onSelect,
   onCancel,
+  selectedDataset,
+  setSelectedDataset,
+  setIndexPattern,
```

Comment:
ideally index pattern is a subset of datasets types. so it's little bit confusing to have a registered data type of `Index Patterns` here https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/query/query_string/dataset_service/lib/index_pattern_type.ts. so ideally datasets selector shouldn't need to care about index patterns